synapse/module_api/__init__.py

```diff
@@ -854,11 +854,8 @@ def complete_sso_login(
             client_redirect_url: The URL to which to offer to redirect the user (or to
                 redirect them directly if whitelisted).
         """
-        self._auth_handler._complete_sso_login(
-            registered_user_id,
-            "<unknown>",
-            request,
-            client_redirect_url,
+        raise Exception(
+            "ModuleApi.complete_sso_login is deprecated in favor of ModuleApi.complete_sso_login_async"
```

Comment:
This isn't really true, since the method is not just deprecated anymore, it's been fully removed.
Also, if we're going to raise an exception, why not remove the method entirely?